modules/packages/RangeChunk.chpl

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright 2004-2016 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ * 
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * 
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module RangeChunk {
+  enum RemElems {
+    Thru,
+    Pack,
+    Mod
+  }
+  use RemElems;
+  use BoundedRangeType;
+
+  /*
+   * Range-based interface, iterator and query.
+   * idx in query is zero based.
+   */
+
+  iter chunks(
+    r: range(?RT, bounded, ?S),
```

Comment:
Sorry about that, went into Javascript mode and didn't think to change it for some reason.